src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs

```diff
@@ -0,0 +1,123 @@
+(ns status-im.contexts.wallet.wallet-connect.session-proposal.view
+  (:require
+    [quo.core :as quo]
+    [quo.foundations.colors :as colors]
+    [quo.theme]
+    [react-native.core :as rn]
+    [status-im.common.floating-button-page.view :as floating-button-page]
+    [status-im.contexts.wallet.wallet-connect.session-proposal.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn- dapp-metadata
+  []
+  (let [proposer  (rf/sub [:wallet-connect/session-proposer])
+        metadata  (-> proposer :metadata)
+        icon      (-> metadata :icons first)
+        dapp-name (-> metadata :name)
+        url       (-> metadata :url)]
```

Comment:
 in most of those threading is not needed

Reply:
you could simply destructure here, right at line 14